test/e2e/e2e.go

```diff
@@ -189,12 +190,54 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 
 })
 
+type CleanupActionHandle *int
+
+var cleanupActionsLock sync.Mutex
+var cleanupActions = map[CleanupActionHandle]func(){}
+
+// AddCleanupAction installs a function that will be called in the event of the
+// whole test being terminated.  This allows arbitrary pieces of the overall
+// test to hook into SynchronizedAfterSuite().
+func AddCleanupAction(fn func()) CleanupActionHandle {
+	p := new(int)
```

Comment:
I could make the handle be int and return a pointer to handle, but that's
not right.  I added a cast (frankly I am not sure why it works without the
cast).

On Tue, Feb 9, 2016 at 9:34 AM, Isaac Hollander McCreery <
notifications@github.com> wrote:

> In test/e2e/e2e.go
> <https://github.com/kubernetes/kubernetes/pull/20771#discussion_r52343173>
> :
>
> > @@ -189,12 +190,54 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
> >
> >  })
> >
> > +type CleanupActionHandle *int
> > +
> > +var cleanupActionsLock sync.Mutex
> > +var cleanupActions = map[CleanupActionHandle]func(){}
> > +
> > +// AddCleanupAction installs a function that will be called in the event of the
> > +// whole test being terminated.  This allows arbitrary pieces of the overall
> > +// test to hook into SynchronizedAfterSuite().
> > +func AddCleanupAction(fn func()) CleanupActionHandle {
> > +	p := new(int)
>
> Is there a way to type p better here? It'd be better to use
> CleanupActionHandle.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20771/files#r52343173>.
>